rust/worker/src/execution/orchestration/hnsw.rs

```diff
@@ -221,7 +225,21 @@ impl HnswQueryOrchestrator {
         }
     }
 
-    async fn hnsw_segment_query(&mut self, ctx: &ComponentContext<Self>) {
+    fn get_disallowed_ids(&self, logs: Chunk<LogRecord>) -> Vec<String> {
+        let mut disallowed_ids = Vec::new();
+        for item in logs.iter() {
+            let log = item.0;
+            let operation_record = &log.record;
+            if operation_record.operation == Operation::Delete
+                || operation_record.operation == Operation::Update
+            {
+                disallowed_ids.push(operation_record.id.clone());
```

Comment:
would be interesting if there is a cheap way to present a mask over the chunk to generate this so we don't have to clone

Reply:
An alternative solution would be creating a new data chunk that masks only the Delete and Update records to be visible then pass that to the lower level. Can the lower level handle that?

Reply:
we could - what do you think is better?

Reply:
No copy of id is needed. The iteration may take longer. This depends on the number of ids and which solution is better depends on the input. I will go with this solution for now based on the assumption that the number of updates and deletes are small.